src/main/java/com/pluang/OnboardingCryptoService/services/PortfolioService.java

```
@@ -0,0 +1,25 @@
+package com.pluang.OnboardingCryptoService.services;
+import com.pluang.OnboardingCryptoService.entities.Portfolio;
+import com.pluang.OnboardingCryptoService.repositories.PortfolioRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class PortfolioService {
+    @Autowired
+    PortfolioRepository portfolioRepository;
+    //update portfolio entry record
+    public Portfolio putPortfolio(Portfolio portfolio) {
+        return portfolioRepository.save(portfolio);
+    }
+    public Portfolio getPortfolio(Integer id) {
+
+        return portfolioRepository.findById(id).orElse(null);
```

If we don't find the portfolio, we should throw a custom exception with 404 status code